README.md

```diff
@@ -123,9 +123,9 @@ how to interpret them.  They're comprised of the following key/value pairs:
     decimal digits of a number.  The byte sequence `0x12 0x34` would translate
     to the decimal value `1234`.  When converting BCD values, treat the
     nibbles 0xA to 0xF as 0 numerically, or a space for display purposes.
-  - `"ch"`: A sequence of 7-bit ASCII characters.  If the JSON file has a 
-    `_char_map` key, use bytes from the NV file as indexes into that string 
-    instead of interpreting them as 7-bit ASCII.
+  - `"ch"`: A null-terminated sequence of 7-bit ASCII characters.  
```

Comment:
`gldneye` uses 
```
        "initials": {
          "start": "0x14ba",
          "encoding": "ch",
          "length": 3
        },
```
without _char_map and they are not 0-terminated

![image](https://github.com/user-attachments/assets/b1b136b6-1b4c-4d27-9bfc-5138560f92cc)

Reply:
As are many others, so I would create a new encoding

Reply:
My intent was that it still be fixed-length, with the possibility of a null terminator shortening the length.  So maybe I should say, "A sequence of 7-bit ASCII characters that may be shortened by a 0x00 null terminator."

Which reminds me -- I don't know whether a SAM game with a 10-letter name will null terminate it or not.  That could be important to know for editing `.nv` files.  I also neglected to add `checksum16` entries to those files.

Reply:
We have to make sure that writing these values will be correct. If the user for example enters a value that is too long it will need to be truncated to length -1 and a `\0` added at the end. For the normal "ch" encoding this is not the case, there we need truncation to length and no `\0`.

eg:

field is 6 chars
"mycoolname"
ch: `mycool`
zero: `myco\0`

It's just a different encoding, mixing them up will cause trouble.

Reply:
Good point.  How do you feel about `string` for null-terminated string that has to fit inside the assigned area?  But we still have two types:

- if it's short, null terminate, but if it fits then do not null terminate
- always null terminate, and length of entry accounts for the null terminator
 
Can you check through your `.nv` files for a SAM game with a 10-letter name that's 10 letters?  I'm curious if it places a 0x00 in the 11th position or not.

Reply:
could also be an extra property on "ch" fields that expects/forces newlines

have not tested 10-letter names yet

Reply:
Proposing a `null` property for `ch`.  Three options:
- `ignore` (default): Current behavior of not doing anything special with null bytes.
- `terminate`: Null byte *required* to terminate the field.  Field length includes a byte for the null-terminator.
- `truncate`: Characters can fill the fixed size of the field, but a null byte truncates the remaining bytes.

Reply:
Have you seen the `truncate` option in the wild?

Reply:
For `truncate`, we'd have to take a look at various platforms that used 3-character initials, but allowed the player to select "END" early when entering initials.  Did all of them just put space (0x20) in the blank spots (WPC95 has blank after GG)?  Or did some of them use 0x00?

But I see those as the only three possibilities.  That 0x00 means nothing, 0x00 can shorten the fixed-length string, or 0x00 is always required.  I could see Stern going either way on 10-letter names.  It will all depend on whether it writes the 0x00 after the name (length=11 and `terminate`) or not (length=10 and `truncate`).  If we document them and code them correctly, they're there when needed.